qutip/qip/device/processor.py

```diff
@@ -700,17 +734,19 @@ def run_state(self, init_state=None, analytical=False, states=None,
         # choose solver:
         if solver == "mesolve":
             solver = mesolve
+            evo_result = mesolve(
+                H=noisy_qobjevo, rho0=init_state,
+                tlist=noisy_qobjevo.tlist, **kwargs)
         elif solver == "mcsolve":
-            solver = mcsolve
+            evo_result = mcsolve(
+                H=noisy_qobjevo, psi0=init_state,
+                tlist=noisy_qobjevo.tlist, **kwargs)
```

Comment:
Why does the solver variable get overridden with a function value in one case but not the other?